test/Libraries/CoreNodesTests/CoreNodeModelsTests/SelectionTests.cs

```diff
@@ -42,6 +42,44 @@ public void TestsSetup()
             testNode = CreateCodeBlockNode();
         }
 
+        [Test]
+        public void ConstructorWithPortsTest()
+        {
+            //Ports for testing
+            PortModel inPort = new PortModel(PortType.Input, testNode,new PortData("input", "input port"));
+            PortModel outPort = new PortModel(PortType.Output, testNode,new PortData("output", "output port"));
+
+            IEnumerable<PortModel> inPorts = new List<PortModel>{inPort};
+            IEnumerable<PortModel> outPorts = new List<PortModel>{outPort};
+
+            //Creates the new selection
+            selection = new SelectionConcrete(
+                SelectionType.Many,
+                SelectionObjectType.Element,
+                "testMessage",
+                "testPrefix",
+                new List<string>{"selection Identifier"},
+                inPorts,
+                outPorts);
+
+            Assert.IsNotNull(selection);
```

Comment:
You `new` anything, you'll expect its value to non-null :)

Reply:
you are right I'll update with better assertions

Reply:
I changed the assertion so that it checks that the ports that where provided as parameters are the ones being used, which is what that constructur is used for